src/main/java/com/bootcamp/demo/restapi/BookingsController.java

```diff
@@ -0,0 +1,135 @@
+        } catch (ExecutionException | InterruptedException | IllegalArgumentException e) {
+            e.printStackTrace();
+            return new ResponseEntity<>(bookingsByUserId, HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    @GetMapping("/getBooking/{id}")
+    @ResponseBody
+    public ResponseEntity<Object> getBooking(@PathVariable(value = "id") final UUID id) {
+        Booking booking = null;
+        try {
+            booking = bookingService.getBookingByID(id);
+            LOGGER.info("GET BOOKING BY ID - API endpoint invoked");
+            return new ResponseEntity<>(booking, HttpStatus.OK);
+        } catch (ExecutionException | InterruptedException | IllegalArgumentException e) {
+            e.printStackTrace();
+            return new ResponseEntity<>(booking, HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    @GetMapping("/showAllBookings")
+    @ResponseBody
+    public ResponseEntity<LinkedHashSet<Booking>> getAllBookings() {
+        LinkedHashSet<Booking> bookings = new LinkedHashSet<>();
+        try {
+            bookings = bookingService.getAllBookings();
+            return new ResponseEntity<>(bookings, HttpStatus.OK);
+        } catch (ExecutionException | InterruptedException e) {
+            e.printStackTrace();
+            return new ResponseEntity<>(bookings, HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    @PutMapping("/delete")
+    public ResponseEntity<Object> deleteBooking(@RequestParam final UUID id) {
+        try {
+            bookingService.deleteBooking(id);
+            LOGGER.info("DELETE BOOKING - API endpoint invoked");
+            return new ResponseEntity<>(HttpStatus.OK);
+        } catch (ExecutionException | InterruptedException | IllegalArgumentException e) {
+            e.printStackTrace();
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    @PutMapping("/update")
+    public ResponseEntity<Object> updateBooking(@RequestParam final UUID id, @RequestParam("start")
+    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)  LocalDateTime endDate, PaymentStatus status) {
+        try {
+            bookingService.updateBooking(id, endDate, status);
+            LOGGER.info("UPDATE BOOKING - API endpoint invoked");
+            return new ResponseEntity<>(HttpStatus.OK);
+        } catch (ExecutionException | InterruptedException | IllegalArgumentException e) {
+            e.printStackTrace();
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    public double endOfBooking(Booking booking, Scooter scooter) throws ExecutionException, InterruptedException {
+        long bookingDuration=0;
+        double totalCostComputed=0.0;
+        if(booking.getEndDate().isAfter(booking.getEndDate())) {
+            bookingDuration = (booking.getEndDate().getMinute() - booking.getStartDate().getMinute());
+
+        }
+        totalCostComputed=booking.getTotalCost(bookingDuration);
```

Comment:
You could both declare & initialize this here, e.g. `final double totalCostComputed`